WordPress/gutenberg
# Add visualizers for padding and margin for all blocks

closes #36418 
closes #40057

## What?

This PR adds "visualizers" when editing padding/margin controls in blocks that support them.

## Why?

The visualizers gives the user clarity of the impact its changes are having on the block.

## How?

Previously we had `BoxControl.Visualizer` components, it was used in the Cover block padding for instance, but it suffered two major issues:

 - It relied on rendering a "div" in the DOM inside the block itself. The problem is that this impacts the layout and style of the blocks (causes layout shifts...), it's not a technique we can use in all blocks. The current PR solves this by making the visualizer work like the block toolbar using a "Popover". It uses a special Popover that tries to mirror the size of the block to cover it.
 - The second issue is that the visualizers relied on block "attribute" changes to support showing them on "hover". This causes issues like creating "undo" levels, making the "post" dirty and keeping a trace in the saved markup as soon as you "hover" a padding control. I elected to remove that entirely and only show the visualizers on "edit" right now. If it's deemed not sufficient, I can look at restoring the hover effect without using the attributes.

**Note**

I'm planning to use the BlockPopover (with cover argument) in #40376 to render the side remove buttons... So this PR is also a requirement for that.

## Testing Instructions

1- Insert a block that supports padding or margin (group, cover...)
2- Tweak the value for these controls
3- Notice the visual effect (blue squares) when changing the value.

## Screenshots or screencast 

https://user-images.githubusercontent.com/272444/164422610-e01d5904-4f52-4aaf-93bd-ea4abdd5a70c.mov
